src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -279,13 +279,62 @@ end
 
 /-- Finite measures yield elements of the `weak_dual` of bounded continuous nonnegative
 functions via `finite_measure.test_against_nn`, i.e., integration. -/
-def to_weak_dual_bounded_continuous_nnreal (μ : finite_measure α) :
+def to_weak_dual_bcnn (μ : finite_measure α) :
   weak_dual ℝ≥0 (α →ᵇ ℝ≥0) :=
 { to_fun := λ f, μ.test_against_nn f,
   map_add' := test_against_nn_add μ,
   map_smul' := test_against_nn_smul μ,
   cont := μ.test_against_nn_lipschitz.continuous, }
 
+lemma to_weak_dual_bcnn_eval' (μ : finite_measure α) (f : α →ᵇ ℝ≥0) :
+  μ.to_weak_dual_bcnn f = μ.test_against_nn f := rfl
+
+lemma to_weak_dual_bcnn_eval (μ : finite_measure α) (f : α →ᵇ ℝ≥0) :
+  μ.to_weak_dual_bcnn f = (∫⁻ x, f x ∂(μ : measure α)).to_nnreal := rfl
```

Comment:
This should be called `to_weak_dual_bcnn_apply`, and `simps` can generate it for you. If you do use `simps`, make sure to do `@[simps apply]` so you *don't* get the `coe_apply` lemma too.

Reply:
Thanks! I had not used `@[simps]` before. This looks very useful for auto-generating boilerplate and consistent naming conventions. I think it will take me some time to use this efficiently, but I am now removing a few boilerplate lemmas and addins the `@[simps apply]`.

The one rewrite lemma that simps doesn't generate is rewriting all the way to an integral. I expected that to be frequently used, but I'm not totally sure, so I'm leaving it out now; it can anyway be achieved by at most two rewrites.

Reply:
I'm not sure I'm getting the simps right --- sorry, this is new to me.

I now tried just tagging the two `to_weak_dual_bcnn`s with `@[simps apply]`; it sounds plausible to me that the autogenerated lemmas are a good simp normal form. With these, the linter complained about the simp lemma `coe_to_weak_dual_bcnn`, so I left it out now.

Reply:
I think you definitely want the `coe` versions, and I think you want them to be simp lemmas. Normally `simps` could just take care of both for you, but in the case of the `to_fun` projection `simps` generates a double coercion lemma, which is why I wrote the coe lemma by hand. 

In this case, I think we might just have to write both lemmas by hand because of the order in which they appear. Notice that the following passes the `simp_nf` linter, whereas generating the second one via `simps` makes it fail.
```lean
@[simp] lemma coe_to_weak_dual_bcnn (μ : finite_measure α) :
  ⇑μ.to_weak_dual_bcnn = μ.test_against_nn := rfl

@[simp] lemma to_weak_dual_bcnn_apply (μ : finite_measure α) (f : α →ᵇ ℝ≥0) :
  μ.to_weak_dual_bcnn f = (∫⁻ x, f x ∂(μ : measure α)).to_nnreal := rfl
```

Reply:
Thank you! I added these for both finite measures and probability measures and removed the simps-tags from the definitions.